### DISCUSSION TITLE
Javascript inside custom module

### DISCUSSION THREAD

**Discussion by chybyk** [OrchardCMS/OrchardCore#15448] [Q&A] [answered]
Hello, I'm new to the OrchardCore.
Can I add my own JavaScript file and register it with the ResourceManager in my custom module?
And then work with it like this in my views:
`ResourceManager.RegisterResource("script", "my-custom-scipt").AtHead();`

I couldn't find any example or tutorial.

I realized that through ResourceManager you can add a script as text, but if I want to add it as a file, where should I place it correctly and specify the path to it in ResourceManager.

Thanks.

**Comment by hishamco** [comment]
I already did something similar in [OrchardCoreContrib.Themes](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Themes), have a look to https://github.com/OrchardCoreContrib/OrchardCoreContrib.Themes/blob/dev/src/OrchardCoreContrib.Themes.Resume/ResourceManagementOptionsConfiguration.cs

**Reply by chybyk** [reply]
As far as I know, the module is a class library and it cannot be done there as in the templates...
It might be easier to make your own template and define your scripts in it.
Thanks for the info!

**Reply by hishamco** [reply]
If you find an issue or something to improve, feel free to file an issue in the related repo

**Comment by weirdyang** [comment, marked as answer]
If I understand correctly, you want to register a script and be able to reference the script file in your templates?
Create a class which inherits from `IConfigureOptions<ResourceManagementOptions>`
```
    public class ResourceManifest : IConfigureOptions<ResourceManagementOptions>
    {
        private static ResourceManagement.ResourceManifest _manifest;

        static ResourceManifest()
        {
            _manifest = new ResourceManagement.ResourceManifest();

            _manifest
                .DefineScript("custom-script") // this will be the name you reference
                .SetUrl("~/OrchardCore.Custom/Custom/js/activity.js");
        }

        public void Configure(ResourceManagementOptions options)
        {
            options.ResourceManifests.Add(_manifest);
        }
    }
```
Register it in your module start up
```
        public override void ConfigureServices(IServiceCollection services)
        {
            services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManifest>();
        }
```
Reference it in your templates
```
 <script asp-name="custom-script" at="Foot"></script>
```

Apologies if I misunderstood

**Reply by chybyk** [reply]
Looks like what I was looking for. Thanks for the example.

**Comment by hishamco** [comment]
Seems the right answer is what I refer to in my sample :)

**Reply by chybyk** [reply]
I marked it as an answer because it was implemented through a module, it seemed to me that it works differently with themes. Sorry if I didn't understand something)
